eng/pipelines/templates/stages/vmr-build.yml

```diff
@@ -89,51 +91,14 @@ stages:
 
   - ${{ if ne(variables['Build.Reason'], 'PullRequest') }}:
```

Comment:
This check feels a bit strange with the isFullBuild parameter.  It is a bit of a coupled system that assumes isFullBuild is only intended to be used outside of a PullRequest.  

Would switching or negating isFullBuild to isLiteBuild help this situation?  I think conditions in this template would make more sense.

Reply:
The issue is that the first 3 legs are also used by the full build. By switching to `isLiteBuild`, the first three legs become exclusive to the lite pipeline. For that reason, I don't think that switching to `isLiteBuild` makes sense.

Reply:
Given that the first three legs must run in the full build, I don't think that using isLiteBuild would make sense. That is, if we used isLiteBuild, it would exclude the first three legs from the full build. An alternative solution is to move the legs that run for a full build (and full build only), outside of checking if the build reason is pull request, since the full pipeline will never be triggered by a PR.

The other solution is to create another template for the first three stages called vmr-build-lite. This template would be called by ci-lite.yml, and it would be inherited by vmr-build for the full pipeline.

Reply:
> That is, if we used isLiteBuild, it would exclude the first three legs from the full build.

I wasn't thinking a check would be added for the first three legs, they would always run.  The difference would be the other legs would be excluded when running in isLiteBuild.  eg. ${{ if eq(parameters.isLightBuild, false) }}:

The value of this is the afore mentioned `${{ if ne(variables['Build.Reason'], 'PullRequest') }}:` would make sense simple because there is not isFullBuild parameter.  IDK if this makes sense or not.

Reply:
Changed to isLiteBuild in https://github.com/dotnet/installer/pull/17370/commits/bf697aa5cab242f80871e0db905b3d85ba24dedf